static/js/formatters-internal.js

```diff
@@ -321,13 +321,41 @@ export function image(simpleOrComplexImage = {}, desiredSize = '200x', atLeastAs
     throw new Error(`Object of type boolean expected. Got ${typeof atLeastAsLarge}.`);
   }
 
-  let desiredWidth, desiredHeight;
-  let desiredDims = desiredSize.split('x');
+  const isEuImage = image.url.includes('eu.mktgcdn.com');
 
-  const [urlWithoutExtension, extension] = _splitStringOnIndex(image.url, image.url.lastIndexOf('.'));
+  const dynamicUrl = isEuImage
+    ? _getEuImageDynamicUrl(image, desiredSize, atLeastAsLarge)
+    : _getUsImageDynamicUrl(image.url, desiredSize, atLeastAsLarge);
+
+  return Object.assign(
+    {},
+    image,
+    {
+      url: dynamicUrl.replace('http://', 'https://')
+    }
+  );
+}
+
+/**
+ * Given a US image url, returns the dynamic url.
+ *
+ * @param {string} imageUrl Image's url (e.g.
```

Comment:
@EmilyZhang777 it's somewhat hard for me to tell from this diff, but are we repeating the logic to deal with the dimensions separately for the EU and US functions?  If we are, could we refactor to separate those concerns so that the dimension logic is properly encapsulated?

Reply:
I actually tried to refactor this part so they can be reused. But we'll need to handle atLeastAsLarge differently between EU vs US, and it made things unnecessarily complicated that way.

Reply:
is the difference that the EU url has format options at the end as params rather than as `<width>x<height>` dimensions?

Reply:
Yes. EU also handles atLeastAsLarge differently as the dynl/dynm domain only supports US images

Reply:
Ah ok, that's unfortunate!  Do the US urls not let you use the EU syntax? that seems like an odd choice.

Reply:
US images can use dynl/dynm.mktgcdn.com to handle the different fit settings, which is not available for EU. More details here: https://hitchhikers.yext.com/docs/content/image-storing-and-serving/
I CAN switch US images off of them. But I think it's preferred to use them if available.

Reply:
Ok sounds good